include/xtensor-python/pycontainer.hpp

```diff
@@ -29,12 +30,22 @@
 
 #include "xtensor/xcontainer.hpp"
 
-#define import_numpy() { if (_import_array() < 0) {PyErr_Print(); PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import"); } }
-
 namespace xt
 {
 
     /**
+     * Import the numpy Python module.
+     */
+    inline void import_numpy()
+    {
+        if (_import_array() < 0)
+        {
+            PyErr_Print();
+            PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import");
+        }
```

Comment:
You can put the body of the function inside `#ifndef NO_IMPORT_ARRAY` - `#endif`.
You could also split the declaration / definition.